check offline/insiders VSIX

elahehrashedi opened the pull request:
Add check to ensure offline/insiders VSIX is installed on the correct platform
add Github download link to the correct platform
bugfix: https://github.com/microsoft/vscode-cpptools/issues/5119

Commit:
check offline/insiders VSIX

Commit:
check offline/insiders VSIX

Commit:
draft

Commit:
draft

bobbrow commented on Extension/src/main.ts:
```diff
@@ -48,6 +48,24 @@ export async function activate(context: vscode.ExtensionContext): Promise<CppToo
+    let installedPlatform: string = await util.getInstalledBinaryPlatform();
+    if (process.platform !== installedPlatform) {
+        let helpURL: string = "https://github.com/Microsoft/vscode-cpptools/releases";
+        errMsg = localize("native.binaries.not.supported", "The installed VSIX is incompatible with {0}. Download the {0} VSIX from: {1}.", process.platform, helpURL);
```
I would like to avoid using "VSIX" in the message since most non-MSFT folks don't refer to the extension that way.  I think a simple message that says something like: "The {0} version of the extension appears to be installed. Please download and install the {1} version of the extension." would be better.  Putting the URL in the message is unnecessary because the button will take you there.

bobbrow replied:
Or "The {0} version of the extension is incompatible with your OS"

bobbrow commented on Extension/src/common.ts:
```diff
@@ -534,6 +534,33 @@ export function checkInstallLockFile(): Promise<boolean> {
 
+/** Get the platform that the installed binaries belong to.*/
+export function getInstalledBinaryPlatform(): Promise<string> {
+    return new Promise(async (resolve, reject) => {
```
Let's avoid creating explicit Promises.  If you mark this function as `async` you can use `await` directly and there's no need to call `resolve` or `reject`.  Just return the result if it succeeds and throw a `new Error` if it fails.

pieandcakes committed:
Adding the arm/arm64 packages for mono runtime (#5064)

* Adding the arm/arm64 packages for mono runtime

Adding the arm/arm64 packages for the mono runtime. This is to enable
remote-ssh extension debugging.

* Undo activate change

* Fix lint error

* Remove vsce dependency

csigs committed:
Localization - Translated Strings (#5339)

Co-authored-by: DevDiv Build Lab - Dev14 <dlab14@microsoft.com>

Commit:
add offline url, modify warning message

Commit:
add offline url, modify warning message

Commit:
insiders update/downgrade improvements (#5309)

* add test, no warning yet

* check for the assets, add more test

* addd test for Default channel

* fix spelling srrors

* update Default chanel tests

Co-authored-by: Elaheh Rashedi <elrashed@microsoft.com>
Co-authored-by: Sean McManus <seanmcm@microsoft.com>

Commit:
Merge branch 'elrashed/5119_2' of https://github.com/Microsoft/vscode-cpptools into elrashed/5119_2

Commit:
modify warning message
